### PR TITLE
Feature/dp filter library

### PR DESCRIPTION
**Pull request by Matt-Guest**
### What

Migrated the observation package to the `dp-filter` repo

### How to review

Run unit tests, manual run the export process

### Who can review

anyone

### CODE REVIEW

File: filter/store.go [hunk 1]
```diff
@@ -23,6 +23,25 @@ type Store struct {
...
 
+// Downloads represents a list of file types possible to download
+type FilterOuput struct {
+	FilterID         string                         `json:"filter_id,omitempty"`
+	InstanceID       string                         `json:"instance_id"`
+	State            string                         `json:"state,omitempty"`
+	Downloads        *Downloads                     `json:"downloads,omitempty"`
+}
+
+// DimensionFilter represents an object containing a list of dimension values and the dimension name
```
Comment (CarlHembrough) [inline filter/store.go]: Comment needs fixing here

File: filter/store.go [hunk 2]
```diff
@@ -23,6 +23,25 @@ type Store struct {
 	httpClient         HTTPClient
 }
 
+// Downloads represents a list of file types possible to download
```
Comment (CarlHembrough) [inline filter/store.go]: Comment needs fixing here

File: filter/store.go [hunk 3]
```diff
@@ -52,13 +71,13 @@ func (store *Store) PutCSVData(filterJobID string, fileURL string, size int64) e
...
 
 	// Add the CSV file to the filter job, the filter api will update the state when all formats are completed
-	putBody := &observation.Filter{
-		Downloads: &observation.Downloads{
-			CSV: &observation.DownloadItem{
-				Size: strconv.FormatInt(size, 10),
-				URL:  fileURL,
-			},
+	putBody := &FilterOuput{
+	Downloads: &Downloads{
```
Comment (CarlHembrough) [inline filter/store.go]: run gofmt

Comment (Matt-Guest) [reply filter/store.go]: ran go fmt 👍

File: filter/store.go (hunk 2 above)
Comment (Matt-Guest) [reply filter/store.go]: updated comment

File: filter/store.go (hunk 1 above)
Comment (Matt-Guest) [reply filter/store.go]: updated comment

**Review by CarlHembrough — APPROVED** [review]

**Pull request merged**